### PR TITLE
Feature/eslint

### PR DESCRIPTION
**Pull request by shellyear**
Implementing eslint incrementally in this PR, considering discussion [here](https://github.com/kbss-cvut/record-manager-ui/issues/103)

### CODE REVIEW

**Comment by blcham** [comment]
Not sure if i was supposed to check this but if `npm run validate` i get following error:
image

**Comment by shellyear** [comment]
> Not sure if i was supposed to check this but if `npm run validate` i get following error: image

I forgot to push the changes yesterday to the remote, now it should work

File: src/components/user/User.jsx [hunk 1]
```diff
@@ -1,3 +1,4 @@
+/* eslint-disable react/no-string-refs */
```
Comment (blcham) [inline src/components/user/User.jsx:1]: See comment above referenced as [1].

File: .eslintrc.cjs [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  extends: ["plugin:react/recommended"],
+  parserOptions: {
+    requireConfigFile: false,
+    babelOptions: {
+      presets: ["@babel/preset-react"],
+    },
+  },
+  plugins: ["react"],
+  rules: {}, // add specific rules here
+  ignorePatterns: ["dist", "tests", ".eslintrc.cjs"],
```
Comment (blcham) [inline .eslintrc.cjs:12]: Should we mention "tests" in #110 so we do not forget about it ?

File: package.json [hunk 3]
```diff
@@ -45,12 +48,19 @@
     "dotenv": "^8.2.0",
     "dotenv-safe": "^8.2.0",
     "enzyme": "^3.11.0",
+    "eslint": "^8.57.0",
+    "eslint-config-airbnb": "^19.0.4",
```
Comment (blcham) [inline package.json:52]: I believe we should remove those and also other deps that are not relevant.

File: src/components/HorizontalInput.jsx [hunk 4]
```diff
@@ -35,6 +36,8 @@ export default class HorizontalInput extends React.Component {
   }
 
   focus() {
+    /* eslint-disable react/no-find-dom-node */
```
Comment (blcham) [inline src/components/HorizontalInput.jsx:39]: I suggest to create ticket for all issues like this one. [1]

File: src/components/Input.jsx [hunk 5]
```diff
@@ -1,3 +1,4 @@
+/* eslint-disable react/no-find-dom-node */
```
Comment (blcham) [inline src/components/Input.jsx:1]: See comment above referenced as [1].

File: src/components/institution/Institution.jsx [hunk 6]
```diff
@@ -1,3 +1,4 @@
+/* eslint-disable react/no-string-refs */
```
Comment (blcham) [inline src/components/institution/Institution.jsx:1]: See comment above referenced as [1].

File: src/components/login/Login.jsx [hunk 7]
```diff
@@ -59,7 +60,13 @@ class Login extends React.Component {
       return (
         <span key={browser.name}>
           {browser.linkMobile && browser.linkDesktop ? (
-            <a href={deviceIsMobile() ? browser.linkMobile : browser.linkDesktop} target="_blank" key="0">
+            // Using target="_blank" without rel="noreferrer" (which implies rel="noopener") is a security risk in older browsers: see https://mathiasbynens.github.io/rel-noopener/#recommendations
```
Comment (blcham) [inline src/components/login/Login.jsx:63]: I would make separate issue for that.

File: src/components/user/PasswordChange.jsx [hunk 8]
```diff
@@ -1,3 +1,4 @@
+/* eslint-disable react/no-string-refs */
```
Comment (blcham) [inline src/components/user/PasswordChange.jsx:1]: See comment above referenced as [1].

File: src/components/user/User.jsx [hunk 9]
```diff
@@ -106,6 +108,7 @@ class User extends React.Component {
           style={{ margin: "0 0.3em 0 0" }}
           variant="primary"
           size="sm"
+          // eslint-disable-next-line react/no-string-refs
```
Comment (blcham) [inline src/components/user/User.jsx:111]: See comment above referenced as [1].

File: vite.config.js [hunk 10]
```diff
@@ -1,14 +1,20 @@
...
 
 export default defineConfig({
   base: "",
   root: "",
   envPrefix: "RECORD_MANAGER_",
-  plugins: [react()],
+  plugins: [
+    react(),
+    eslintPlugin({
+      cache: false, // disable eslint cache to avoid conflicts
```
Comment (blcham) [inline vite.config.js:15]: why do we need it?

**Review by blcham — COMMENTED** [review]
Lets discuss it in a while, you can answer some of the questions and some harder to explain leave for the meeting.

File: vite.config.js (hunk 10 above)
Comment (shellyear) [reply vite.config.js:15]: to ensure that eslint performs linting checks without relying on cached results

File: package.json (hunk 3 above)
Comment (blcham) [reply package.json:52]: Please remove everything that you do not use.

**Comment by blcham** [comment]
@shellyear please create  ticket for warnings that it generates !

File: src/components/user/User.jsx (hunk 1 above)
Comment (blcham) [reply src/components/user/User.jsx:1]: Done in https://github.com/kbss-cvut/record-manager-ui/issues/111

File: src/components/HorizontalInput.jsx (hunk 4 above)
Comment (blcham) [reply src/components/HorizontalInput.jsx:39]: https://github.com/kbss-cvut/record-manager-ui/issues/111

File: src/components/institution/Institution.jsx (hunk 6 above)
Comment (blcham) [reply src/components/institution/Institution.jsx:1]: https://github.com/kbss-cvut/record-manager-ui/issues/111

File: src/components/user/PasswordChange.jsx (hunk 8 above)
Comment (blcham) [reply src/components/user/PasswordChange.jsx:1]: https://github.com/kbss-cvut/record-manager-ui/issues/111

File: src/components/Input.jsx (hunk 5 above)
Comment (blcham) [reply src/components/Input.jsx:1]: https://github.com/kbss-cvut/record-manager-ui/issues/111

File: package.json (hunk 3 above)
Comment (blcham) [reply package.json]: done

File: src/components/login/Login.jsx (hunk 7 above)
Comment (blcham) [reply src/components/login/Login.jsx:63]: Done in https://github.com/kbss-cvut/record-manager-ui/issues/112

**Comment by shellyear** [comment]
> * [ ]  create  ticket for warnings that eslint generates !
> *it means not the potential errors(if we use some plugins like airbnb), but the actual ones, that I see when running npm run lint(mainly css related warnings)

**Comment by shellyear** [comment]
> @shellyear please
> 
> * [ ]  create  ticket for warnings that eslint generates !
> * [ ]  make sure that application works

The ticket with warnings is [here](https://github.com/kbss-cvut/record-manager-ui/issues/113)
The application works fine, with warnings defined in the mentioned [issue]((https://github.com/kbss-cvut/record-manager-ui/issues/113))

**Comment by blcham** [comment]
Merging, congrats 🥳🥳🥳

**Comment by blcham** [comment]
@LaChope this PR was merged, there are 48 files changed mostly only additions, so I suggest that you rebase your work ASAP so you do not have to merge to much later

**Pull request merged**